PKPD/data/abstractData.py

```diff
@@ -0,0 +1,13 @@
+from abc import ABC, abstractmethod
+
+class AbstractData(ABC):
+    def __init__(self):
+        self.datafile = None
+
+    def read_file(self):
+        # read data from a csv (or other) file
+        raise NotImplementedError
+
+    def set_datafile(self):
```

Comment:
Would put this in the constructor or as an argument to read_file ?

Reply:
Hey Michael, 

We've based these abstract classes off of what we did last week, trying to build the classes and get used to the different systems. So all the functions in here are what we think we will need from all we've done so far. This is adaptable as well, so if we realise something needs changing when building, we will.

Reply:
> Would put this in the constructor or as an argument to read_file ?

What would that look like in an abstract class? Could we just remove the set_datafile method and when we make the concrete data class add something to the __init__ to set self.datafile to a string containing the filepath?

Reply:
Yeah you can define a default __init__ in an abstract class!
(E.g. https://github.com/pints-team/pints/blob/master/pints/_log_pdfs.py#L83-L106 )